frontend-react/src/pages/submissions/FilterContext.tsx

```diff
@@ -0,0 +1,174 @@
+import React, {
+    createContext,
+    PropsWithChildren,
+    useContext,
+    useEffect,
+    useState,
+} from "react";
+import { useResource } from "rest-hooks";
+
+import usePaginator, { PaginationController } from "../../utils/UsePaginator";
+import SubmissionsResource from "../../resources/SubmissionsResource";
+import { GlobalContext } from "../../components/GlobalContextProvider";
+
+/* Convenient type aliases */
+type SortOrder = "ASC" | "DESC";
+type PageSize = 10 | 25 | 50 | 100;
+type FilterTypes = string | SortOrder | PageSize;
+
+type StateUpdater = (filter: FilterName, val?: FilterTypes) => void;
+type StateClearer = () => void;
+
+export enum FilterName {
+    START_RANGE = "start-range",
+    END_RANGE = "end-range",
+    CURSOR = "cursor",
+    SORT_ORDER = "sort-order",
+    PAGE_SIZE = "page-size",
+}
+
+/* Object containing state from a FilterContext */
+export interface FilterState {
+    startRange: string;
+    endRange: string;
+    cursor: string;
+    sortOrder: SortOrder;
+    pageSize: PageSize;
+}
+
+/* Shape of the object provided to Submissions pages */
+interface ISubmissionFilterContext {
+    filters: FilterState;
+    contents: SubmissionsResource[];
+    updateFilter: StateUpdater;
+    clear: StateClearer;
+    paginator?: PaginationController;
+}
+
+/* This is a definition of the context shape, NOT the payload delivered */
+export const SubmissionFilterContext = createContext<ISubmissionFilterContext>({
+    filters: {
+        startRange: "",
+        endRange: "",
+        cursor: "",
+        sortOrder: "DESC",
+        pageSize: 10,
+    },
+    contents: [],
+
+    /* Placeholders. Given console.logs to prevent critical smells in
+     * SonarCloud.
+     */
+    updateFilter: (filter: FilterName, val?: FilterTypes) => {
+        console.log("to please SonarCloud");
+    },
+    clear: () => {
+        console.log("to please SonarCloud");
+    },
+});
+
+/* This component handles filter state and API call refreshing for
+ * tables that wish to use filtering. Additionally, this integrates
+ * the Paginator hook.
+ */
+const FilterContext: React.FC<any> = (props: PropsWithChildren<any>) => {
```

Comment:
Could this be exported out from a utility file?

Reply:
Like an `index.ts` for the directory? I like that idea!

Reply:
Yep, that's the idea that came to mind 🧠